### PR TITLE
epsilon value to norm function in aimnet2

### PR DESCRIPTION
**Pull request by chrisiacovella**
## Pull Request Summary
As discussed in #390 aimnet2 was failing when training with forces.  I tracked this down to be where we take the norm of the vector contributions.  Adding in a small epsilon value (1e-8) seems to correct the issue. 

This also includes a few additional example scripts, e.g., to calculate the self-energy of a dataset and plotting a dataset. 

Note: CI was failing due to an update to pytorch lightning.  See #392 .  Basically, we now also need to pass weights_only to various pytorch lightning functions we are calling. 

### Key changes
Notable points that this PR has either accomplished or will accomplish.
 - [x] fix issue causing crashes when training with force

### Associated Issue(s)
 - [x] #390 
 - [x] #392 

## Pull Request Checklist
 - [x] Issue(s) raised/addressed and linked
 - [x] Includes appropriate unit test(s)
 - [x] Appropriate docstring(s) added/updated
 - [x] Appropriate .rst doc file(s) added/updated
 - [x] PR is ready for review

### CODE REVIEW

**Comment by chrisiacovella** [comment]
Running any real training tests (with forces) that would help to detect these errors is going to be too expensive for CI and require too much memory.  I think I should create a separate test (that will be skipped on CI) to do these checks...they will not necessarily be that expensive to run on a machine with GPU locally.   While I was seeing failure during the first epoch with the full dataset, I'm not sure if the n=1000 test set will also fail so quickly (or fail at all).  I need to do some additional local tests on the unmodified version to ensure failure with the test set (so that any actual testing with the patched version is actually meaningful).

File: modelforge/potential/aimnet2.py [hunk 1]
```diff
@@ -435,10 +435,14 @@ def calculate_vector_contributions(
         avf_v_sum.index_add_(0, idx_j, avf_v)  # Shape: (number_of_atoms, H, 3)
 
         # Compute the norm over the last dimension (vector components)
+        # note, we need to add a very small epsilon value of 1e-8 to ensure stability
         vector_contributions = torch.norm(
-            avf_v_sum, dim=-1
+            avf_v_sum + 1e-8, dim=-1
         )  # Shape: (number_of_atoms, H)
```
[the remark this answers is not in the document]
Comment (chrisiacovella) [reply modelforge/potential/aimnet2.py]: Switching to add an epsilon to the normalized result doesn't fix the stability issue which is associated with the calculation of the norm of very small values.

**Review by MarshallYan — APPROVED** [review]
